comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +24,172 @@ public static bool IsDirectory(string path)
 
 		// Get the path and size (in bytes) of the smallest file below the given directory
 		public static Tuple<string, long> GetSmallestFile(string directory)
 		{
-			throw new NotImplementedException();
+			// Gets every element in 'directory', this includes files or sub-directories.
+			string[] elementsInDirectory = Directory.GetFileSystemEntries(directory);
+			Tuple<string, long> smallestFile = new Tuple<string, long>(String.Empty, 0);
+			Tuple<string, long> smallestFileInSubDir;
+			long currentFileSize;
+
+			foreach (string element in elementsInDirectory)
+			{
+				// If the element is a directory then 'GetSmallestFile' is called recursively.
+				// This means all sub-directories are also checked.
+				if (IsDirectory(element))
+				{
+					smallestFileInSubDir = GetSmallestFile(element);
+					// The ' smallestFile.Item2 == 0' is needed to stop this function always returning a value of zero. 
+					if (smallestFile.Item2 > smallestFileInSubDir.Item2 || smallestFile.Item2 == 0)
+					{
+						smallestFile = smallestFileInSubDir;
+					}
+				}
+				else
+				{
+					currentFileSize = GetFileSize(element);
+					// The ' smallestFile.Item2 == 0' is needed to stop this function always returning a value of zero. 
+					if (smallestFile.Item2 > currentFileSize || smallestFile.Item2 == 0)
+					{
+						smallestFile = new Tuple<string, long>(element, currentFileSize);
+					}
+				}
+			}
+
+			return smallestFile;
 		}
 
 		// Get the path and size (in bytes) of the largest file below the given directory
+		// This function is functionally similar to 'GetSmallestFile', checking for a larger file size rather than a smaller one.
 		public static Tuple<string, long> GetLargestFile(string directory)
 		{
-			throw new NotImplementedException();
+			// Gets every element in 'directory', this includes files or sub-directories.
+			string[] elementsInDirectory = Directory.GetFileSystemEntries(directory);
+			Tuple<string, long> largestFile = new Tuple<string, long>(String.Empty, 0);
+			Tuple<string, long> largestFileInSubDir;
+			long currentFileSize;
+
+			foreach (string element in elementsInDirectory)
+			{
+				// If the element is a directory then 'GetLargestFile' is called recursively.
+				// This means all sub-directories are also checked.
+				if (IsDirectory(element))
+				{
+					largestFileInSubDir = GetLargestFile(element);
+					if (largestFile.Item2 < largestFileInSubDir.Item2)
+					{
+						largestFile = largestFileInSubDir;
+					}
+				}
+				else
+				{
+					currentFileSize = GetFileSize(element);
+					if (largestFile.Item2 < currentFileSize)
+					{
+						largestFile = new Tuple<string, long>(element, currentFileSize);
+					}
+				}
+			}
+
+			return largestFile;
 		}
 
 		// Get all files whose size is equal to the given value (in bytes) below the given directory
-		public static IEnumerable<string> GetFilesOfSize(string directory, long size)
+		public static List<string> GetFilesOfSize(string directory, long size)
```

Comment:
The `List<T>` class implements the `IEnumerable<T>` interface, so you needn't have changed the return type of the function